investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java

```diff
@@ -45,235 +47,221 @@ public class ProcessInvestigationDataUtil {
+
+            for(JsonNode node : actIdsJsonArray) {
+                int actIdSeq = node.get("act_id_seq").asInt();
+                String typeCode = node.get(TYPE_CD).asText();
+                String rootExtension = node.get("root_extension_txt").asText();
+
+                if(typeCode.equals("STATE") && actIdSeq == 1) {
+                    investigationTransformed.setInvStateCaseId(rootExtension);
+                }
+                if(typeCode.equals("CITY") && actIdSeq == 2) {
+                    investigationTransformed.setCityCountyCaseNbr(rootExtension);
+                }
+                if(typeCode.equals("LEGACY") && actIdSeq == 3) {
+                    investigationTransformed.setLegacyCaseId(rootExtension);
                 }
             }
-            else {
-                logger.info("ActIds array is null.");
-            }
+        } catch (IllegalArgumentException ex) {
+            logger.info(ex.getMessage(), "ActIds");
         } catch (Exception e) {
             logger.error("Error processing Act Ids JSON array from investigation data: {}", e.getMessage());
         }
     }
 
-    private void transformObservationIds(String observationNotificationIds, InvestigationTransformed investigationTransformed, ObjectMapper objectMapper) {
+    private void transformObservationIds(String observationNotificationIds, InvestigationTransformed investigationTransformed) {
         try {
-            JsonNode investigationObservationIdsJsonArray = parseJsonArray(observationNotificationIds, objectMapper);
+            JsonNode investigationObservationIdsJsonArray = parseJsonArray(observationNotificationIds);
             InvestigationObservation investigationObservation = new InvestigationObservation();
             List<Long> observationIds = new ArrayList<>();
 
-            if(investigationObservationIdsJsonArray != null) {
-                for(JsonNode node : investigationObservationIdsJsonArray) {
-                    String sourceClassCode = node.get("source_class_cd").asText();
-                    String actTypeCode = node.get("act_type_cd").asText();
-                    Long publicHealthCaseUid = node.get("public_health_case_uid").asLong();
-                    investigationKey.setPublicHealthCaseUid(publicHealthCaseUid);
-
-                    if(sourceClassCode.equals("OBS") && actTypeCode.equals("PHCInvForm")) {
-                        investigationTransformed.setPhcInvFormId(node.get("source_act_uid").asLong());
-                    }
-
-                    if(sourceClassCode.equals("OBS") && actTypeCode.equals("LabReport")) {
-                        investigationObservation.setPublicHealthCaseUid(publicHealthCaseUid);
-                        observationIds.add(node.get("source_act_uid").asLong());
-                    }
+            for(JsonNode node : investigationObservationIdsJsonArray) {
+                String sourceClassCode = node.path("source_class_cd").asText();
```

Comment:
At some of point of time, do you think it will be good to convert this to a POJO model class